torch/distributed/_tensor/__init__.py

```diff
@@ -1,13 +1,21 @@
 # Copyright (c) Meta Platforms, Inc. and affiliates
+import copy
 from typing import Optional, Sequence
 
 # Import all builtin dist tensor ops
 import torch
 import torch.distributed._tensor.ops
+import torch.distributed._tensor.random as random
 from torch.distributed._tensor._utils import compute_local_shape
 from torch.distributed._tensor.api import distribute_module, distribute_tensor, DTensor
 from torch.distributed._tensor.device_mesh import DeviceMesh, mesh_resources
-from torch.distributed._tensor.placement_types import Placement, Replicate, Shard
+from torch.distributed._tensor.placement_types import (
+    DTensorSpec,
+    Placement,
+    Replicate,
+    Shard,
+)
+from torch.fx.passes.shape_prop import TensorMetadata
```

Comment:
can we make sure the additional imports (DTensorSpec, tensormetadata) be conditional imports? we should make them into the `elif` branch, the reason is that we should be careful to add new imports to `__init__.py` as that's not sth we want to expose to user.